### ISSUE TITLE
[MDEP-198] Maven dependency go-offline doesn't download surefire-api jar

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1168]
**[Cimballi](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cimballi)** opened **[MDEP-198](https://issues.apache.org/jira/browse/MDEP-198?redirect=false)** and commented

I have this in my pom.xml :

```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-surefire-plugin</artifactId>
  <version>2.4.3</version>
  ...
</plugin>
```

And if I run the goal dependency:go-offline, the surefire-api-2.4.3.jar is not downloaded and so the project cannot be run offline. The surefire plugin and the surefire-boot jars are downloaded.

---

2 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Falko Modler](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=famod)** commented

I have a similar problem:

plexus-utils-1.1.jar is **not** downloaded, but plexus-utils-1.1.pom **is** downloaded.

Probably even more jars are missing. This renders go-offline completely useless for my use case.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Works flawlessly with `maven-dependency-plugin:2.10`:

```
D:\Entwicklung\Projekte\maven-filtering>mvn dependency:go-offline
[INFO] Scanning for projects...
Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/2.4.3/maven-surefire-plugin-2.4.3.jar
Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-surefire-plugin/2.4.3/maven-surefire-plugin-2.4.3.jar (23 KB at 48.7 KB/sec)
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Apache Maven Filtering 3.1.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] >>> maven-dependency-plugin:2.10:go-offline (default-cli) > :resolve-plugins @ maven-filtering >>>
[INFO]
[INFO] --- maven-dependency-plugin:2.10:resolve-plugins (resolve-plugins) @ maven-filtering ---
[INFO] Plugin Resolved: maven-checkstyle-plugin-2.15.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-descriptor-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-impl-2.3.jar
[INFO]     Plugin Dependency Resolved: maven-shared-resources-2.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-decoration-model-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.4.jar
[INFO]     Plugin Dependency Resolved: doxia-integration-tools-1.6.jar
[INFO]     Plugin Dependency Resolved: plexus-component-annotations-1.5.5.jar
[INFO]     Plugin Dependency Resolved: plexus-resources-1.0.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.20.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.21.jar
[INFO]     Plugin Dependency Resolved: plexus-velocity-1.1.8.jar
[INFO]     Plugin Dependency Resolved: checkstyle-6.1.1.jar
[INFO]     Plugin Dependency Resolved: velocity-1.5.jar
[INFO]     Plugin Dependency Resolved: commons-collections-3.2.1.jar
[INFO] Plugin Resolved: maven-project-info-reports-plugin-2.9.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-impl-2.3.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-repository-metadata-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-settings-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-dependency-tree-2.2.jar
[INFO]     Plugin Dependency Resolved: doxia-integration-tools-1.6.jar
[INFO]     Plugin Dependency Resolved: maven-shared-jar-1.2.jar
[INFO]     Plugin Dependency Resolved: maven-common-artifact-filters-1.4.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-2.7.jar
[INFO]     Plugin Dependency Resolved: wagon-ssh-2.7.jar
[INFO]     Plugin Dependency Resolved: wagon-file-2.7.jar
[INFO]     Plugin Dependency Resolved: wagon-http-lightweight-2.7.jar
[INFO]     Plugin Dependency Resolved: maven-scm-api-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-manager-plexus-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-hg-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-perforce-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-starteam-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-svn-commons-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-svnexe-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-cvs-commons-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-cvsexe-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-cvsjava-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-git-commons-1.9.2.jar
[INFO]     Plugin Dependency Resolved: maven-scm-provider-gitexe-1.9.2.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.6.jar
[INFO]     Plugin Dependency Resolved: doxia-core-1.6.jar
[INFO]     Plugin Dependency Resolved: doxia-decoration-model-1.6.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.6.jar
[INFO]     Plugin Dependency Resolved: plexus-i18n-1.0-beta-10.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.20.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.21.jar
[INFO]     Plugin Dependency Resolved: plexus-container-default-1.0-alpha-9-stable-1.jar
[INFO]     Plugin Dependency Resolved: commons-validator-1.4.0.jar
[INFO]     Plugin Dependency Resolved: commons-lang-2.6.jar
[INFO] Plugin Resolved: apache-rat-plugin-0.11.jar
[INFO]     Plugin Dependency Resolved: apache-rat-core-0.11.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: doxia-core-1.2.jar
[INFO]     Plugin Dependency Resolved: doxia-decoration-model-1.2.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.2.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.2.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-testing-harness-1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.8.jar
[INFO] Plugin Resolved: maven-clean-plugin-3.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-shared-utils-3.0.0.jar
[INFO] Plugin Resolved: maven-resources-plugin-2.7.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-settings-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-monitor-2.2.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.15.jar
[INFO]     Plugin Dependency Resolved: maven-filtering-1.2.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.19.jar
[INFO] Plugin Resolved: maven-surefire-plugin-2.4.3.jar
Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-booter/2.4.3/surefire-booter-2.4.3.jar
Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-booter/2.4.3/surefire-booter-2.4.3.jar (262 KB at 518.0 KB/sec)
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.6.jar
[INFO]     Plugin Dependency Resolved: surefire-booter-2.4.3.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-1.5.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-toolchain-1.0.jar
[INFO] Plugin Resolved: maven-deploy-plugin-2.8.2.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.15.jar
[INFO] Plugin Resolved: maven-compiler-plugin-3.5.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-toolchain-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-shared-utils-3.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-shared-incremental-1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-api-2.7.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-manager-2.7.jar
[INFO]     Plugin Dependency Resolved: plexus-compiler-javac-2.7.jar
[INFO] Plugin Resolved: maven-jar-plugin-3.0.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-3.0.2.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-3.1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO] Plugin Resolved: maven-remote-resources-plugin-1.5.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-monitor-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-settings-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-resolver-1.0.jar
[INFO]     Plugin Dependency Resolved: maven-common-artifact-filters-1.4.jar
[INFO]     Plugin Dependency Resolved: maven-filtering-1.1.jar
[INFO]     Plugin Dependency Resolved: plexus-resources-1.0-alpha-7.jar
[INFO]     Plugin Dependency Resolved: plexus-interpolation-1.12.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-1.5.15.jar
[INFO]     Plugin Dependency Resolved: velocity-1.7.jar
[INFO] Plugin Resolved: maven-site-plugin-3.5.1.jar
[INFO]     Plugin Dependency Resolved: maven-reporting-exec-1.2.jar
[INFO]     Plugin Dependency Resolved: maven-core-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-model-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-settings-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-settings-builder-3.0.jar
[INFO]     Plugin Dependency Resolved: maven-archiver-2.5.jar
[INFO]     Plugin Dependency Resolved: sisu-inject-plexus-1.4.2.jar
[INFO]     Plugin Dependency Resolved: plexus-archiver-2.9.jar
[INFO]     Plugin Dependency Resolved: plexus-i18n-1.0-beta-7.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.20.jar
[INFO]     Plugin Dependency Resolved: plexus-classworlds-2.2.3.jar
[INFO]     Plugin Dependency Resolved: doxia-sink-api-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-module-xhtml-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-module-apt-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-module-xdoc-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-module-fml-1.7.jar
[INFO]     Plugin Dependency Resolved: doxia-module-markdown-1.7.jar
[INFO]     Plugin Dependency Resolved: servlet-api-2.5.jar
[INFO]     Plugin Dependency Resolved: doxia-decoration-model-1.7.1.jar
[INFO]     Plugin Dependency Resolved: doxia-site-renderer-1.7.1.jar
[INFO]     Plugin Dependency Resolved: doxia-integration-tools-1.7.1.jar
[INFO]     Plugin Dependency Resolved: wagon-provider-api-1.0.jar
[INFO]     Plugin Dependency Resolved: jetty-6.1.25.jar
[INFO]     Plugin Dependency Resolved: jetty-util-6.1.25.jar
[INFO]     Plugin Dependency Resolved: commons-lang3-3.4.jar
[INFO]     Plugin Dependency Resolved: commons-io-1.4.jar
[INFO] Plugin Resolved: maven-install-plugin-2.5.2.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-manager-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: commons-codec-1.6.jar
[INFO]     Plugin Dependency Resolved: maven-shared-utils-0.4.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.15.jar
[INFO] Plugin Resolved: maven-enforcer-plugin-1.4.1.jar
[INFO]     Plugin Dependency Resolved: maven-artifact-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.2.1.jar
[INFO]     Plugin Dependency Resolved: maven-core-2.2.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.22.jar
[INFO]     Plugin Dependency Resolved: commons-lang-2.3.jar
[INFO]     Plugin Dependency Resolved: enforcer-api-1.4.1.jar
[INFO]     Plugin Dependency Resolved: enforcer-rules-1.4.1.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-testing-harness-1.3.jar
[INFO] Plugin Resolved: plexus-component-metadata-1.7.1.jar
[INFO]     Plugin Dependency Resolved: plexus-container-default-1.7.1.jar
[INFO]     Plugin Dependency Resolved: plexus-component-annotations-1.7.1.jar
[INFO]     Plugin Dependency Resolved: plexus-utils-3.0.20.jar
[INFO]     Plugin Dependency Resolved: qdox-2.0-M2.jar
[INFO]     Plugin Dependency Resolved: jdom2-2.0.6.jar
[INFO]     Plugin Dependency Resolved: maven-plugin-api-2.0.9.jar
[INFO]     Plugin Dependency Resolved: maven-model-2.0.9.jar
[INFO]     Plugin Dependency Resolved: maven-project-2.0.9.jar
[INFO]     Plugin Dependency Resolved: plexus-cli-1.6.jar
[INFO]     Plugin Dependency Resolved: commons-cli-1.2.jar
[INFO]     Plugin Dependency Resolved: asm-all-5.1.jar
[INFO]
[INFO] <<< maven-dependency-plugin:2.10:go-offline (default-cli) < :resolve-plugins @ maven-filtering <<<
[INFO]
[INFO] --- maven-dependency-plugin:2.10:go-offline (default-cli) @ maven-filtering ---
[INFO] Resolved: maven-core-3.0.jar
[INFO] Resolved: maven-model-3.0.jar
[INFO] Resolved: maven-settings-3.0.jar
[INFO] Resolved: maven-settings-builder-3.0.jar
[INFO] Resolved: maven-repository-metadata-3.0.jar
[INFO] Resolved: maven-artifact-3.0.jar
[INFO] Resolved: maven-plugin-api-3.0.jar
[INFO] Resolved: maven-model-builder-3.0.jar
[INFO] Resolved: maven-aether-provider-3.0.jar
[INFO] Resolved: aether-impl-1.7.jar
[INFO] Resolved: aether-spi-1.7.jar
[INFO] Resolved: aether-api-1.7.jar
[INFO] Resolved: aether-util-1.7.jar
[INFO] Resolved: sisu-inject-plexus-1.4.2.jar
[INFO] Resolved: sisu-inject-bean-1.4.2.jar
[INFO] Resolved: sisu-guice-2.1.7-noaop.jar
[INFO] Resolved: plexus-classworlds-2.2.3.jar
[INFO] Resolved: plexus-component-annotations-1.6.jar
[INFO] Resolved: plexus-sec-dispatcher-1.3.jar
[INFO] Resolved: plexus-cipher-1.4.jar
[INFO] Resolved: maven-shared-utils-3.1.0.jar
[INFO] Resolved: commons-io-2.5.jar
[INFO] Resolved: plexus-utils-3.0.24.jar
[INFO] Resolved: plexus-interpolation-1.24.jar
[INFO] Resolved: plexus-build-api-0.0.7.jar
[INFO] Resolved: jsr305-3.0.0.jar
[INFO] Resolved: plexus-build-api-0.0.4-tests.jar
[INFO] Resolved: mockito-core-1.9.5.jar
[INFO] Resolved: hamcrest-core-1.1.jar
[INFO] Resolved: objenesis-1.0.jar
[INFO] Resolved: junit-4.11.jar
[INFO] Resolved: assertj-core-1.7.1.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.965 s
[INFO] Finished at: 2017-04-15T20:50:39+02:00
[INFO] Final Memory: 19M/208M
[INFO] ------------------------------------------------------------------------

D:\Entwicklung\Projekte\maven-filtering>mvn -v
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
Maven home: D:\Entwicklung\Programme\apache-maven-3.3.9\bin\..
Java version: 1.8.0_112, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.8.0_112\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "dos"
```
